modules/sort/test/spec.coffee

```diff
@@ -108,3 +141,52 @@ describe 'hx-sort tests', ->
         if supportsOptions()
           array.sort(hx.sort.localeCompare('sv')).should.eql(['a', 'ä', 'e', 'é', 'è', 'z'])
           array.sort(hx.sort.localeCompare('de')).should.eql(['a', 'e', 'é', 'è', 'z', 'ä'])
+
+  describe 'using localeCompare potentially with Intl.Collator. ', localeCompareTests
+
+  describe 'using localeCompare without Intl.Collator. ', runWithoutCollator(localeCompareTests)
+
+  compareNullsLastTests = ->
+    it 'Strings should return the right value', ->
+      hx.sort.compareNullsLast('a', 'b').should.equal(-1)
+      hx.sort.compareNullsLast('b', 'a').should.equal(1)
+      hx.sort.compareNullsLast('a', 'a').should.equal(0)
+
+    it 'undefined should always be sorted to the end', ->
+      hx.sort.compareNullsLast(undefined, 'a').should.equal(1)
+      hx.sort.compareNullsLast('a', undefined).should.equal(-1)
+      hx.sort.compareNullsLast(undefined, 'z').should.equal(1)
```

Comment:
Apparently, that behaviour is consistent, after running the same test suite twice. It wasn't consistent before and probably isn't completely consistent now since James hinted that Intl.Collator tries to be quite clever. But I think for now it is good enough and at least now we can test new cases that come up